### DISCUSSION TITLE
Can SnowDDL CLI accept placeholder variable values?

### DISCUSSION THREAD

**Discussion by minghungcho** [littleK0i/SnowDDL#5] [Ideas]
ex. In my view, I would reference an object in a different database. SELECT * FROM ${{sales_db}}.dbo.revenue. And define sales_db in placeholder.yaml.

However, this value is different for different release pipeline. Ex. there is only non-production and product database of sales_db. so I want to reference sales_db_noprod for all non-production uses (my_db_dev, my_db_qa), but reference sales_db from production databsae (my_db_prod).

With current SnowDDL, I need to write a python code to update placeholder.yaml based on my current pipeline before invoking SnowDDL CLI. If there is an option similar to --vars in schemachage, I can dynamically substitute placeholder values. This is specially useful when invoking SnowDDL as part of Azure DevOp when environment variables are defined in [Azure Release Pipeline](https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/cli/pipeline-variable-group-secret-nonsecret-variables?view=azure-devops)

SNOWDDL --placeholder {"sales_db": "sales_db_noprod", "wh_size":"large", "bucket_name": "$aws_s3_uri"}

**Comment by littleK0i** [comment]
Please check the latest version `0.5.0`. I've added CLI parameter `--placeholder-values`, which works exactly as you described.

```
snowddl ... --placeholder-values='{"WH_SIZE": "LARGE", "DEFAULT_ENFORCE_LENGTH": false}' --apply-unsafe apply
```

A few extra notes:

- placeholder names are uppercased automatically;
- data types of placeholder values are important (!) supported types are: boolean, float, integer, string;
